llvm/lib/TableGen/TableGenBackend.cpp

```diff
@@ -59,4 +60,8 @@ void llvm::emitSourceFileHeader(StringRef Desc, raw_ostream &OS) {
   printLine(OS, Prefix, ' ', Suffix);
   printLine(OS, "\\*===", '-', "===*/");
   OS << '\n';
+
+  // Print the path of source file
+  if (!Record.getInputFilename().empty())
+    OS << "// Generated from: " << Record.getInputFilename() << "\n\n";
```

Comment:
I think this could be enabled if the base path would be given from the build system.

Reply:
Good question! I was thinking it would give just the base file name, not the full path to the file: https://github.com/llvm/llvm-project/blob/aa8f5e6156821aec1fed595f2e13d69655ec3311/llvm/lib/TableGen/Main.cpp#L115 -- that looks like it will generate whatever was given on the command line, which could very well be a full path.